File: Python Problems/inOrderTrav.py
# Given the root of a binary tree, return the inorder traversal of its nodes' values. 



class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def inorderTraversalRecursive(root):
    result = []
    
    def traverse(node):
        if not node:
            return
        
        traverse(node.left)   # Traverse left subtree
        result.append(node.val)  # Visit current node
        traverse(node.right)  # Traverse right subtree
    
    traverse(root)
    return result

File: JavaScript Problems/inorderTrav.js
// Given the root of a binary tree, return the inorder traversal of its nodes' values. in javascript


function inorderTraversalRecursive(root) {
    const result = [];
    
    function traverse(node) {
        if (!node) return;
        
        traverse(node.left); // Traverse left subtree
        result.push(node.val); // Visit current node
        traverse(node.right); // Traverse right subtree
    }
    
    traverse(root);
    return result;
}